setup.cfg

```diff
@@ -43,7 +43,7 @@ install_requires =
     pandas
     pystac>=1.0.0,<2
     toolz
-    xarray ~= 0.20.1
+    xarray
```

Comment:
Is there a reason to drop the version requirement? Personally I like to have some sort of gating for libraries to guard against API-breaking version bumps.

Reply:
Three main reasons

- `xarray` is very common requirement, so lot's of other libs depend on it
- `0.20.1` is still very new version, like a month old
- We don't rely on super fancy features of this lib

Since our usage of `xarray` is fairly pedestrian and that part can be easily worked around I think it's better to support older versions even if it means not using some convenience feature. But, I agree, having a lower bound further back in time would be useful, not sure how to go about finding that.

Reply:
`xarray ~= 0.19` or whatever, and then see if someone reports breakages with an old version? If I understand the `~=` syntax correctly, `~= 0.19` will allow v0.20 (and anything until v1.0).

Reply:
I'm reading [these docs](https://www.python.org/dev/peps/pep-0440/#compatible-release), and I think`~=0.19` won't allow `0.20` and higher, or maybe it won't allow `1.0` and higher. Future proofing dependencies is hard. I feel like having "future version bound" on a dependency is only justified if you are very closely coupled with that package or that package is known to cause major breakage from one release to another. Trade off the way I see it

- Don't add future bound and risk your code breaking when dependencies change in the future in incompatible ways
- Add future bound and risk causing dependency hell even though your package would work just fine with the new version

So I would argue to add lower bound only, (i.e. `>=0.19`) and not bother attempting to future proof against future changes in `xarray` library. Once version `1.0` of xarray is out, then having `<2` in there would make more sense.

Reply:
> I think~=0.19 won't allow 0.20 and higher

`~= 0.19` == `>= 0.19 && == 0.*`, so it would allow `0.20`.

> Add future bound and risk causing dependency hell even though your package would work just fine with the new version

I'd argue that this is ok, becuase it's good guard against such a big change as a `1.0` release (especially for a library that has been shown to be liberal with minor releases in the v0.* chain). But that's a style thing, I think `>= 0.19, < 2` would work too.

Reply:
you mean, `>= 0.19, <1`? I can add that in the next PR. I do want to make a release soon, to get last release with `odc.index.` present but deprecated out, then next one won't have `odc.index.*` there.

Reply:
> you mean, >= 0.19, <1

It sounded like you were more worried about a major bump to `2` putting us in dependency hell, so `>= 0.19, <2` is exactly that scenario.